lib/token.js

```diff
@@ -0,0 +1,285 @@
+'use strict';
+
+const types = require('./types');
+const util = require('util');
+
+const _Murmur3TokenType = types.dataTypes.getByName('bigint');
+const _RandomTokenType = types.dataTypes.getByName('varint');
+const _OrderedTokenType = types.dataTypes.getByName('blob');
+
+/**
+ * Represents a token on the Cassandra ring.
+ *
+ * @property value The raw value of the token.  Type depends on
+ *                 partitioner implementation. 
+ * 
+ */
+class Token {
+  constructor(value, tokenizer) {
+    this.value = value;
+    Object.defineProperty(this, '_tokenizer', { value: tokenizer, enumerable: false});
+  }
+
+  /**
+   * @returns {{code: number, info: *|Object}} The type info for the
+   *                                           type of the value of the token.
+   */
+  getType() {
+    throw new Error('You must implement a getType function for this Token instance');
+  }
+
+  /**
+   * @returns {*} The raw value of the token.
+   */
+  getValue() {
+    return this.value;
+  }
+
+  /**
+   * Encodes the underlying value into a buffer that can be
+   * used for routing.
+   * 
+   * @param {Encoder} encoder Encode for encoding to buffer.
+   * @returns {Buffer} buffer data for the token.
+   */
+  toBuffer(encoder) {
+    return encoder.encode(this.getValue(), this.getType());
+  }
+
+  toString() {
+    return this._tokenizer.stringify(this.getValue());
+  }
+
+  /**
```

Comment:
It could be useful to provide an `inspect()` method (used by log providers / console.log / util.inspect()) to have a human-readable string, that returns something like `this.constructor.name + ' { ' + this.toString() + ' }'`.

Reply:
ah yes, that would work great, the current output isn't currently too readable.